nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaLogManager.java

```diff
@@ -0,0 +1,215 @@
+        int partitions = kUtils.getNumberOfPartitions(getTopicName(partition.name()));
+        if (partition.partition() >= partitions) {
+            throw new IllegalArgumentException("Partition out of bound " + partition + " max: " + partitions);
+        }
+    }
+
+    public Properties getProducerProperties() {
+        return producerProperties;
+    }
+
+    public Properties getConsumerProperties() {
+        return consumerProperties;
+    }
+
+    @Override
+    public void close() throws Exception {
+        super.close();
+        if (kUtils != null) {
+            kUtils.close();
+        }
+    }
+
+    @Override
+    public boolean supportSubscribe() {
+        return !disableSubscribe;
+    }
+
+    @Override
+    protected <M extends Externalizable> LogTailer<M> doSubscribe(String group, Collection<String> names, RebalanceListener listener) {
+        return KafkaLogTailer.createAndSubscribe(prefix, names, group, (Properties) consumerProperties.clone(), listener);
+    }
+
+    protected static Properties normalizeConsumerProperties(Properties consumerProperties) {
+        Properties ret;
+        if (consumerProperties != null) {
+            ret = (Properties) consumerProperties.clone();
+        } else {
+            ret = new Properties();
+        }
+        ret.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+        ret.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.BytesDeserializer");
+        ret.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        ret.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        ret.remove(DISABLE_SUBSCRIBE_PROP);
+        return ret;
+    }
+
+    protected Properties normalizeProducerProperties(Properties producerProperties) {
+        Properties ret;
+        if (producerProperties != null) {
+            ret = (Properties) producerProperties.clone();
+        } else {
+            ret = new Properties();
+        }
+        ret.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+        ret.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.BytesSerializer");
+        ret.remove(DEFAULT_REPLICATION_FACTOR_PROP);
+        return ret;
+    }
+
+    @Override
+    public List<LogLag> getLagPerPartition(String name, String group) {
+        LogLag[] ret;
+        Properties props = (Properties) consumerProperties.clone();
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, group);
+        try (KafkaConsumer<String, Bytes> consumer = new KafkaConsumer<>(props)) {
+            List<TopicPartition> topicPartitions = consumer.partitionsFor(getTopicName(name)).stream()
+                    .map(meta -> new TopicPartition(meta.topic(), meta.partition()))
+                    .collect(Collectors.toList());
+            ret = new LogLag[topicPartitions.size()];
```

Comment:
Use an `ArrayList` instead of an array. No need to declare `ret` outside of the `try`, as it can be returned from inside the `try` block too.